mephisto/abstractions/providers/prolific/prolific_utils.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+from omegaconf import DictConfig
+
+from mephisto.abstractions.crowd_provider import ProviderArgs
+from mephisto.utils.logger_core import get_logger
+
+PROLIFIC_BUDGET = 100000.0
+PROLIFIC_CONFIG_DIR = '~/.prolific/'
+PROLIFIC_CONFIG_PATH = os.path.join(PROLIFIC_CONFIG_DIR, 'credentials')
+
+logger = get_logger(name=__name__)
+
+
+def get_prolific_api_key() -> Union[str, None]:
+    credentials_path = os.path.expanduser(PROLIFIC_CONFIG_PATH)
+    if os.path.exists(credentials_path):
+        with open(credentials_path, 'r') as f:
+            api_key = f.read().strip()
+            return api_key
+    return None
+
+
+PROLIFIC_API_KEY = os.environ.get('PROLIFIC_API_KEY', None) or get_prolific_api_key()
+
+
+def check_credentials(*args, **kwargs) -> bool:
+    """Check whether API KEY is correct"""
+    # TODO (FB-3): Implement
+    pass
+
+
+def setup_credentials(
+    profile_name: str, register_args: Optional[ProviderArgs],
+) -> bool:
+    if not os.path.exists(os.path.expanduser(PROLIFIC_CONFIG_DIR)):
+        os.mkdir(os.path.expanduser(PROLIFIC_CONFIG_DIR))
+
+    with open(os.path.expanduser(PROLIFIC_CONFIG_PATH), 'w') as f:
+        f.write(register_args.api_key)
```

Comment:
Note - one user may be working with _different_ prolific accounts at the same time. For this purpose, we may need to separate out the credentials into separate files (by the `profile_name`) or create a yaml/something to handle this.